### PR TITLE
Add build script to generate artifacts

### PR DESCRIPTION
**Pull request by VijayanB**
Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

### Description
build.sh will generate artifacts using goreleaser.
  
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-cli/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -16,24 +23,29 @@ builds:
...
+      - goos: linux
+        goarch: 386
 archives:
   -
-    name_template: '{{ .ProjectName }}_{{ .Os }}_{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}'
+    name_template: '{{ .ProjectName }}-{{ .Version }}-{{ .Os }}-{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}'
     replacements:
-      darwin: Darwin
-      linux: Linux
-      windows: Windows
```
Comment (vamshin) [inline .goreleaser.yml:30]: Why is Linux And Windows removed?

Comment (VijayanB) [reply .goreleaser.yml:30]: This is replacement section for name. Go uses term Darwin for macOS, but we refer as macos, hence I used it . If we don’t include any other term then default will be used, in this case windows/linux

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

File: .goreleaser.yml [hunk 2]
```diff
@@ -16,24 +23,29 @@ builds:
...
-      386: i386
-      amd64: x86_64
+      darwin: macos
+      386: x86
+      amd64: x64
+    files:
+      - none*
```
Comment (jmazanec15) [inline .goreleaser.yml:40]: Whats the purpose of this?

File: release/build.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#####################################################
+#      Generating Artifacts opensearch-cli                      #
+#                                                   #
+#####################################################
+
+
+echo 'Generating artifacts'
+goreleaser --snapshot --skip-publish --rm-dist
+
+rm -rf dist/opensearch-cli_*
```
Comment (jmazanec15) [inline release/build.sh:27]: Why are we deleting these here?

File: .goreleaser.yml (hunk 2 above)
Comment (VijayanB) [reply .goreleaser.yml:40]: By default, this tool adds README and License file. This is discussed [here](https://github.com/goreleaser/goreleaser/issues/602) and added hack for now

File: release/build.sh (hunk 3 above)
Comment (VijayanB) [reply release/build.sh:27]: This tool generated more than what we wanted I couldn't figure out a way to disable generation of those unwanted files. Hence, deleting the unwanted files. I will add this as comment too.

**Review by jmazanec15 — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**